### PR TITLE
[SDK] Improve host name handling

### PR DESCRIPTION
**Pull request by numerology**
Part of #3280 

Automatic stripping http(s) prefix and redundant '/', and throwing warnings when users only provide the backend id.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/3287)

### CODE REVIEW

**Comment by rmgogogo** [comment]
fix https://github.com/kubeflow/pipelines/issues/3280

File: sdk/python/kfp/_client.py [hunk 1]
```diff
@@ -113,6 +113,11 @@ def __init__(self, host=None, client_id=None, namespace='kubeflow', other_client
 
   def _load_config(self, host, client_id, namespace, other_client_id, other_client_secret):
     config = kfp_server_api.configuration.Configuration()
+
+    host = host or ''
+    # Preprocess the host endpoint to prevent some common user mistakes.
+    host = host.lstrip(r'(https|http)://').rstrip('/')
```
Comment (Ark-kun) [inline sdk/python/kfp/_client.py]: I vaguely remember there were scenarios where this had to be an actual URI with non-empty path.
Something like `http://1.2.3.4/something/#pipeline`

Comment (numerology) [reply sdk/python/kfp/_client.py]: Do you mean we have to reserve the prefix `http://`?

**Pull request merged**